tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,251 @@ fn generate_synthetic(
                     ),
                 )
                 .into();
-                first_transaction.append_instruction(mint);
+                first_transaction.push_instruction(mint);
             }
         }
     }
 
     Ok(genesis)
 }
+
+/// Use `Kagami` to sign genesis block.
+#[derive(ClapArgs, Clone, Debug)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+#[command(group = ArgGroup::new("format").required(false))]
+pub struct SignArgs {
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: crypto::AlgorithmArg,
+    /// Private key (in hex string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
+    /// Path to private key to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_file: Option<PathBuf>,
+    /// Public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_string: Option<PublicKey>,
+    /// Path to public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_file: Option<PathBuf>,
+    /// Path to json-serialized keypair
+    #[clap(long, short, group = "private_key", group = "public_key")]
+    keypair_file: Option<PathBuf>,
+    /// Unique id of blockchain
+    #[clap(long)]
+    chain_id: ChainId,
+    /// Path to genesis json file
+    #[clap(long, short)]
+    genesis_file: PathBuf,
+    /// Output signed genesis config as a hex string
+    #[clap(long, default_value_t = true, group = "format")]
+    hex: bool,
+    /// Encode signed genesis block with SCALE (it is only supported with file output)
+    #[clap(long, default_value_t = false, group = "format")]
+    scale: bool,
```

Comment:
Hm, I think, we can simplify here a little bit. We can replace both `--hex` and `--scale` with a single flag - `--binary`. With `--out-file`, it will write SCALE bytes into the given file, without - will print the same SCALE bytes in a hex format to stdout.